### PR TITLE
Export concrete arena types.

### PR DESCRIPTION
**Pull request by lthibault**

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -384,30 +384,29 @@ type Arena interface {
...
 
-type singleSegmentArena []byte
+// SingleSegmentArena is a simple arena comprising an expanding buffer.
+type SingleSegmentArena []byte
 
 // SingleSegment returns a new arena with an expanding single-segment
 // buffer.  b can be used to populate the segment for reading or to
 // reserve memory of a specific size.  A SingleSegment arena does not
 // return errors unless you attempt to access another segment.
 func SingleSegment(b []byte) Arena {
```
Comment (zenhack) [inline message.go:394]: You probably want to change the return type of this function to match, yeah?

**Review by zenhack — CHANGES_REQUESTED** [review]
A couple things in-line, but mostly this looks good.

File: message.go (hunk 1 above)
Comment (lthibault) [reply message.go]: Fixed

File: message.go [hunk 2]
```diff
@@ -384,30 +384,29 @@ type Arena interface {
 	Allocate(minsz Size, segs map[SegmentID]*Segment) (SegmentID, []byte, error)
 }
 
-type singleSegmentArena []byte
+// SingleSegmentArena is a simple arena comprising an expanding buffer.
+type SingleSegmentArena []byte
```
[the remark this answers is not in the document]
Comment (lthibault) [reply message.go:391]: Done.

**Review by zenhack — APPROVED** [review]

**Pull request merged**